R/rectangle.R

```diff
@@ -426,15 +434,15 @@ vec_to_wide <- function(x, col, names_sep = NULL) {
   }
 
   if (is.data.frame(x)) {
-    as_tibble(map(x, list))
+    new_data_frame(lapply(x, list))
   } else if (vec_is(x)) {
     if (is.list(x)) {
-      x <- purrr::compact(x)
-      x <- map(x, list)
+      x <- vec_slice(x, list_sizes(x) != 0)
```

Comment:
Maybe worth pulling out into a local `compact` helper with a comment explaining the perf benefit? Or maybe vctr should provide `list_compact()`?

Reply:
I pulled it into a compact helper. I also created a `vctrs` issue (https://github.com/r-lib/vctrs/issues/1395) but according to lionel it is out of scope for `vctrs`, at least for now.